src/app/components/driver-list/driver-list.component.ts

```diff
@@ -17,92 +18,89 @@ import { environment } from '../../../environments/environment';
  */
 export class DriverListComponent implements OnInit {
 
-  location : string = 'Morgantown, WV';
+  location : string = '';
   mapProperties :{};
   availableCars : Array<any> = [];
   drivers : Array<any> = [];
-
-
-  @ViewChild('map',null) mapElement: any;
+  availableDrivers;
+  currentUserID: number; // sessionStorage 'userid' value
+  
+  @ViewChild('map', null) mapElement: any;
   map: google.maps.Map;
 
-  constructor(private http: HttpClient,private userService: UserService) { }
+  constructor(private http: HttpClient, private userService: UserService, private carService: CarService) { }
 
   ngOnInit() {
     this.drivers = [];
-
-    this.userService.getRidersForLocation1(this.location).subscribe(
-      res => {
-           //console.log(res);
-           res.forEach(element => {
-              this.drivers.push({
-                   'id': element.userId,
-                 'name': element.firstName+" "+element.lastName,
-               'origin':element.hCity+","+element.hState, 
-                'email': element.email, 
-                'phone':element.phoneNumber
-              });
-          });
-      });
-    
-    this.getGoogleApi();
-
-    this.sleep(2000).then(() => {
-      this.mapProperties = {
-         center: new google.maps.LatLng(Number(sessionStorage.getItem("lat")), Number(sessionStorage.getItem("lng"))),
-         zoom: 15,
-         mapTypeId: google.maps.MapTypeId.ROADMAP
-      };
-      this.map = new google.maps.Map(this.mapElement.nativeElement, this.mapProperties);
-      //get all routes 
-      this.displayDriversList(this.location, this.drivers);
-      //show drivers on map
-      this.showDriversOnMap(this.location, this.drivers);
-    });
+    this.currentUserID = +sessionStorage.getItem('userid');
+
+    // sets the location to the driver's home address
+    this.userService.getDriverById(this.currentUserID).subscribe(
+      (currentUser) => {
+
+        let currentAddress = currentUser.homeAddress;
+        this.location = currentAddress.street + ", " + currentAddress.city + ", " + currentAddress.state;
```

Comment:
This leaves out the apartment/address 2 value in address

Reply:
currentAddress.street also includes the apt number